{
  "comments": [
    {
      "unresolved": true,
      "key": {
        "uuid": "f4a3af7b_05dab4ee",
        "filename": "src/chromiumos/tast/common/hwsec/cryptohome_mount_info.go",
        "patchSetId": 8
      },
      "lineNbr": 53,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "initialise in var statement:\n\n\n```\nvar (\n  // matches a path to vault under /home/shadow.\n  // Example: \"/home/.shadow/118c4648065f5cd3660e17a53533ec7bc924d01f/vault\" \n  shadowRegexp \u003d regexp.MustCompile(`^/home/\\.shadow/[^/]*/vault$`)\n\n  // matches a path to /dev/*.\n  // Example: \"/dev/sda1\"\n  devRegexp \u003d regexp.MustCompile(`^/dev(/[^/]*){1,2}$`)\n\n  // matches a path to /dev/loop\\d+.\n  // Example: \"/dev/loop0\"\n  devLoopRegexp \u003d regexp.MustCompile(`^/dev/loop[0-9]+$`)\n)\n```",
      "range": {
        "startLine": 37,
        "startChar": 0,
        "endLine": 53,
        "endChar": 1
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "877d940e_89e57a79",
        "filename": "src/chromiumos/tast/common/hwsec/cryptohome_mount_info.go",
        "patchSetId": 8
      },
      "lineNbr": 111,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "we cannot just ignore \"err\", you must explicitly check for expected error to be able to return \"-1, nil\".\n\nFor example, you pass context into c.runner.Run and it might expire, so you will get \"context deadline exceeded\" error which is not what you expected",
      "range": {
        "startLine": 111,
        "startChar": 13,
        "endLine": 111,
        "endChar": 16
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "2bb6184c_c13654aa",
        "filename": "src/chromiumos/tast/common/hwsec/cryptohome_mount_info.go",
        "patchSetId": 8
      },
      "lineNbr": 136,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "is it expected to get more than 4 fields?",
      "range": {
        "startLine": 136,
        "startChar": 5,
        "endLine": 136,
        "endChar": 20
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "1fcf68d2_76d0b228",
        "filename": "src/chromiumos/tast/common/hwsec/cryptohome_mount_info.go",
        "patchSetId": 8
      },
      "lineNbr": 194,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "interesting that you just check whether returned error is nil or not. I think it makes sense to return bool instead\n\nsorry that didn\u0027t catch it before",
      "range": {
        "startLine": 194,
        "startChar": 76,
        "endLine": 194,
        "endChar": 81
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "bc35814d_241dc884",
        "filename": "src/chromiumos/tast/local/bundles/cros/hwsec/cryptohome_mount.go",
        "patchSetId": 8
      },
      "lineNbr": 49,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "Shouldn\u0027t it be Fatal instead? does it makes sense to try remove vault after error from unmount?\n\ns.Error won\u0027t stop execution of the function",
      "range": {
        "startLine": 49,
        "startChar": 5,
        "endLine": 49,
        "endChar": 10
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "cff931bf_896233a9",
        "filename": "src/chromiumos/tast/local/bundles/cros/hwsec/cryptohome_mount.go",
        "patchSetId": 8
      },
      "lineNbr": 85,
      "author": {
        "id": 1306673
      },
      "writtenOn": "2021-03-05T11:48:01Z",
      "side": 1,
      "message": "cryptohome.MountVault can return multiple errors, not all of them are expected. Please check explicitly for expected error.",
      "range": {
        "startLine": 85,
        "startChar": 109,
        "endLine": 85,
        "endChar": 119
      },
      "revId": "b03b004c6d138e306f0cce93144e839957feb74a",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    }
  ]
}